File: 1646-kth-missing-positive-number/kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        start = 0
        end = len(arr) - 1
        while (start <= end):
            mid = start + (end-start)//2
            count_missing = arr[mid] - mid - 1
            if count_missing < k:
                start = mid + 1
            else:
                end = mid - 1
        return start + k

File: 1646-kth-missing-positive-number/kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int n=arr.length;
        int left=0, right=n-1;

        while (left<=right) {
            int mid = left + (right-left)/2;
            int miss = arr[mid]-(mid+1);
            if (miss >= k) {
                right = mid-1;
            }
            else {
                left = mid+1;
            }
        }
        return left+k;
    }
}